sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/view/ViewRegistry.java

```diff
@@ -67,21 +72,92 @@ public static ViewRegistryBuilder builder() {
   }
 
   // Matches an instrument selector against an instrument + meter.
-  private static boolean matchesSelector(
+  private boolean matchesSelector(
       InstrumentSelector selector,
       InstrumentDescriptor descriptor,
       InstrumentationScopeInfo meterScope) {
-    return (selector.getInstrumentType() == null
-            || selector.getInstrumentType() == descriptor.getType())
-        && selector.getInstrumentNameFilter().test(descriptor.getName())
-        && matchesMeter(selector, meterScope);
+    if (selector.getInstrumentType() != null
+        && selector.getInstrumentType() != descriptor.getType()) {
+      return false;
+    }
+    if (selector.getInstrumentName() != null
+        && !matchesName(selector.getInstrumentName(), descriptor.getName())) {
+      return false;
+    }
+    return matchesMeter(selector, meterScope);
+  }
+
+  /**
+   * Determine if the {@code instrumentName} matches the {@code namePattern} from {@link
+   * InstrumentSelector#getInstrumentName()}.
+   *
+   * <p>{@code namePattern} may contain the wildcard characters {@code *} and {@code ?} with the
+   * following matching criteria:
+   *
+   * <ul>
+   *   <li>{@code *} matches 0 or more instances of any character
+   *   <li>{@code ?} matches exactly one instance of any character
+   * </ul>
+   */
+  // Visible for testing
+  boolean matchesName(String namePattern, String instrumentName) {
```

Comment:
Ah sorry for not linking this before, can you copy this in?

Oops sorry for not linking to this earlier, can you swap this in?

https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/SamplingRuleApplier.java#L358

Notable minor improvements(?) are optimizing `*`, only iterating once to find wildcard characters, and using `Pattern.quote` instead of huge switch

Reply:
Nice! I've substituted this in, but just use `Predicate` instead of the `Matcher` interface.